components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -15012,6 +15012,53 @@ public String addEndpointRegistryEntry(EndpointRegistryEntry registryEntry) thro
         return uuid;
     }
 
+    /**
+     * Updates Registry Entry
+     *
+     * @param registryEntry EndpointRegistryEntry
+     * @throws APIManagementException if failed to update EndpointRegistryEntry
+     */
+    public void updateEndpointRegistryEntry(EndpointRegistryEntry registryEntry) throws APIManagementException {
+        String query = SQLConstants.UPDATE_ENDPOINT_REGISTRY_ENTRY_SQL;
+        try (Connection connection = APIMgtDBUtil.getConnection();
+             PreparedStatement ps = connection.prepareStatement(query)) {
+            connection.setAutoCommit(false);
+            ps.setString(1, registryEntry.getName());
+            ps.setString(2, registryEntry.getServiceURL());
+            ps.setString(3, registryEntry.getDefinitionType());
+            ps.setString(4, registryEntry.getDefinitionURL());
+            ps.setString(5, registryEntry.getMetaData());
+            ps.setString(6, registryEntry.getServiceType());
+            ps.setBlob(7, registryEntry.getEndpointDefinition().getContent());
+            ps.setString(8, registryEntry.getEntryId());
+            ps.executeUpdate();
```

Comment:
shouldn't we catch the exception in executeUpdate and rollback the connection explicitly. 

See https://stackoverflow.com/questions/3160756/in-jdbc-when-autocommit-is-false-and-no-explicit-savepoints-have-been-set-is-i

Reply:
Fixed in https://github.com/wso2-enterprise/service-catalog/pull/73